src/chess/Command.java

```diff
@@ -0,0 +1,38 @@
+package chess;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+enum Command {
+  HELP,
+  BOARD,
+  RESIGN,
+  ALLPOSSIBLEMOVES,
+  SQUAREPOSSIBLEMOVES,
+  GOMOVE,
+  INVALID;
+
+  static Command parse(String str) {
+    switch (str.toLowerCase()) {
+      case "help":
+        return HELP;
+      case "board":
+        return BOARD;
+      case "resign":
+        return RESIGN;
+      case "moves":
+        return ALLPOSSIBLEMOVES;
+      default:
+        Pattern pattern = Pattern.compile("([a-z]+[0-9]+)(?<secondPos>[a-z]+[0-9]+([a-z]+)?)?");
```

Comment:
You can use `\d` instead of `[0-9]`

Reply:
Last part `([a-z]+)?)` is wring if this part is for "Promotion"

Reply:
`Pattern.compile` might be a bit heavy
When a regular expression is constants you should run this code only once (So you should hold this variable as a private field)